hangman/lib/hangman/game/game.ex

```diff
@@ -0,0 +1,95 @@
+defmodule Hangman.Game do
+  defstruct game_state: :initializing,
+            turns_left: 7,
+            letters: [],
+            used: [],
+            word_to_guess: [],
+            last_guess: ""
+
+  ## [ API ]
+
+  # Creates a game struct with new word and the corresponding blank spaces for
+  # letters
+  def new_game() do
+    word = new_word()
+    letters = create_blank_board(word)
+
+    %Hangman.Game{
+      letters: letters,
+      word_to_guess: word
+    }
+  end
+
+  # Filters Hangman.Game struct to remove the word_to_guess
+  def tally(tally) do
+    tally
+      |> Map.from_struct()
+      |> Enum.reject(fn {key, _value} -> key == :word_to_guess end)
+      |> Enum.into(%{})
+  end
+
+  def make_move(game, guess) do
+    game = game
+      |> check_already_guessed(guess)
+      |> check_guess(guess)
+      |> player_status
+    {game, tally(game)}
+  end
+
+  def check_already_guessed(game, guess) do
+    game.used
+      |> contains_letter?(guess)
+      |> already_guessed(game)
+  end
+
+  def already_guessed(true, game = %Hangman.Game{}),  do: %{game | game_state: :already_guessed}
+  def already_guessed(false, game = %Hangman.Game{}), do: %{game | game_state: :continue}
+
+  def check_guess(game = %Hangman.Game{game_state: :already_guessed}, _), do: game
```

Comment:
Yes! Not many people did the check for already guessed like this. It lets you use a pipeline, which is great.

### +3